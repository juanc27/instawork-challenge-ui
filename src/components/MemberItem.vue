<script setup>
defineProps(['firstName', 'lastName', 'email', 'phone', 'role'])
</script>

<template>
  <div>
    <h2>Name: {{ firstName }} {{ lastName }} <span v-if="role === 'admin'">(admin)</span></h2>
    <h2>Email: {{ email }}</h2>
    <h2>Phone: {{ phone }}</h2>
    <hr class="3px solid #bbb; border-top:"><hr>
  </div>
</template>
