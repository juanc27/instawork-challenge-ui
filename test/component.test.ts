import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import MemberItem from '../src/components/MemberItem.vue'

describe('component of MemberItem.vue', () => {
  it('should render', () => {
    const wrapper = mount(MemberItem, { props: { id: 1, firstName: 'John', lastName: 'Doe', email: 'test@test.com', phone: '1234567890', role: 'admin' } })
    expect(wrapper.text()).toContain('Name: John Doe (admin)')
    expect(wrapper.text()).toContain('Email: test@test.com')
    expect(wrapper.text()).toContain('Phone: 1234567890')
    expect(wrapper.html()).toMatchSnapshot()
  })
})
