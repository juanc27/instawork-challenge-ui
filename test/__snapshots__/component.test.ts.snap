// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`component of MemberItem.vue > should render 1`] = `
"<a href="1">
  <div>
    <p>Name: John Doe <span>(admin)</span></p>
    <p>Email: test@test.com</p>
    <p>Phone: 1234567890</p>
    <hr class="3px solid #bbb; border-top:">
    <hr>
  </div>
</a>"
`;
